src/IonBinaryWriter.ts

```diff
@@ -654,7 +654,7 @@ class IntNode extends LeafNode {
   constructor(writer: LowLevelBinaryWriter, parent: Node, annotations: Uint8Array, private readonly value: number | JSBI) {
     super(writer, parent, IonTypes.INT, annotations);
 
-    if (this.value === 0) {
+    if (this.value == 0) {
```

Comment:
JSBI(0) == "0" -> true
" " == 0 -> true
false == 0 -> true
[] == 0 -> true
We should be checking the input type instead.

Reply:
We should be using `JSBI.equals()` or `JSBI.EQ()` for `===` and `==` respectively.  That said, I agree here that `==` (and its equivalent `JSBI.EQ()`) is not quite right.  What about:

```typescript
this.value === 0 || JsbiSupport.isZero(this.value)
```

Maybe rolling the above into a utility method might be worthwhile.

Reply:
Today, this appears to be only place where we perform a problematic `===` comparison.  I've updated this PR to avoid it altogether via a @zslayton gambit:  `0` is now handled in the else block.  Avoids a lengthy condition, as well as any discussion around where such utility method should reside.

That said, we'd be wise to build a little internal infrastructure around number/JSBI to avoid pitfalls like this in the future, as well as only allocating a JSBI when doing so adds value.